wagtail/admin/views/pages/search.py

```diff
@@ -37,113 +50,147 @@ def page_filter_search(q, pages, all_pages=None, ordering=None):
     return pages, all_pages
 
 
-@vary_on_headers("X-Requested-With")
-@user_passes_test(user_has_any_page_permission)
-def search(request):
-    pages = all_pages = Page.objects.all().prefetch_related("content_type").specific()
-    show_locale_labels = getattr(settings, "WAGTAIL_I18N_ENABLED", False)
-    if show_locale_labels:
-        pages = pages.select_related("locale")
-
-    q = MATCH_ALL
-    content_types = []
-    ordering = None
-
-    if "ordering" in request.GET:
-        if request.GET["ordering"] in [
+class BaseSearchView(PermissionCheckedMixin, BaseListingView):
+    permission_policy = PagePermissionPolicy()
+    any_permission_required = {
+        "add",
+        "edit",
```

Comment:
This should be `"change"` now that #10569 is in.

```suggestion
        "change",
```